### ISSUE TITLE
packaging: Debian packaging is broken yet again

### ISSUE THREAD

**Issue by nekromant** [texane/stlink#482]
Current debian install rules to not take care to install the correct version.h that pretty much make the whole -dev package unusable.  

@texane Just to remember, I'll try to dive and fix it when I have a spare minute.

**Comment by xor-gate** [comment]
I quote:

> Do we really need to duplicate all the file installation rules in debian/rules? I suggest we do as much of the installation as possible with CMakeLists.txt, so that debian/rules would do as much as possible automatically.
> 
> Is there any rationale for building both debug and release builds of stlink when packaging?

**Comment by xor-gate** [comment]
1. I agree. we need to use the install target of cmake.
2. We building a package we should only build release.

**Comment by xor-gate** [comment]
> Use of cmake install would be better in terms of maintenance.
> 
> I am not sure how to tell cmake where to install
> adapt debian/*.install files to copy from cmake install dir, instead of from debian/tmp
> or you can manage to install to debain/tmp with cmake
> cmake should handle multiarch library paths
> Just that tests depends on debug build. The build stage calls make all and all depends on release. The tests stage calls make test and test depends on debug. This is why both where build. I think automated tests where a good thing and should run, but you can disable it by add override_dh_auto_tests: to debian/rules, if you want.

**Comment by xor-gate** [comment]
You can tell cmake to install into `DESTDIR` with `make install`. See https://cmake.org/cmake/help/v3.0/variable/CMAKE_INSTALL_PREFIX.html

**Comment by nekromant** [comment]
@OsterlaD As @texane noted, you can just add DESTDIR= make install. In debian rules you don't have to mention this, as default debian stuff gets the job done already. Basically:
```
override_dh_auto_configure:
        dh_auto_configure -- \
        -DCMAKE_LIBRARY_PATH=$(DEB_HOST_MULTIARCH)
``` 
Is the only thing required and the only thing that I keep around my projects. dpkg does the rest automatically.  
 
@texane For my projects (See http://github.com/nekromant/aura) I add tests for production builds as well, but only those that don't require hardware connected by default. They are just not installed during the 'make install' phase. If dpkg-buildpackage won't find any tests it just won't run any, and since we have a CI I really think we can safely skip tests for packaging.

**Comment by xor-gate** [comment]
@nekromant i'm @xor-gate :=) not @texane :D

**Comment by OsterlaD** [comment]
I thought about this way too.
dh_auto_build will call `make all`, which calls `cmake` at its own.
This is why dh_auto_configure has no effect.
So I removed code from dh_auto_configure.
There is no `make install`.

**Comment by xor-gate** [comment]
We could add `make install`  to the toplevel `Makefile` which just calls `make install` in the release build

**Comment by OsterlaD** [comment]
Sounds good. Maybe I have some time to try it later on.

**Comment by nekromant** [comment]
@xor-gate I'll give it a proper try tomorrow.

**Comment by OsterlaD** [comment]
#484 use of `make install`

Lintian has some suggestions
* rename package to libstlink1, but that conflicts cmake libstlink look-up
* move lib links to dev pkg

I would like to add a postinst script to libstlink to reload udev rules.

**Comment by xor-gate** [comment]
I think this is now completely resolved?

**Comment by nekromant** [comment]
@xor-gate This is more like optional polishing. Postinst would be nice. 
As for lib links, AFAIK I've done that already.

**Comment by xor-gate** [comment]
If you guys don't mind I will close this for now, as this topic was used mostly for discussion. When there specific are debian-related thinks need to be fixed we can open new issues. Thanks all for your effort!

**Comment by Nightwalker-87** [comment]
Finally closed by #485.
